File: 2023/06/6.py
g=lambda t,d:sum(b*(int(t)-b)>int(d)for b in range(int(t)))
r=1
z=[l.split()[1:]for l in open("a")]
for a in zip(*z):r*=g(*a)
print(r,g(*["".join(l)for l in z]))

File: 2023/06/6.clj
(ns aoc.2023.06.6 
  (:require [aoc.common :refer [read-input split-to-ints zip]]
            [clojure.string :as str]))

(defn find-time [[t d]]
  (->> (range)
       (map #(* % (- t %)))
       (drop-while #(<= % d))
       (take-while #(< d %))
       count))

(let [input (read-input)
      races1 (->> input
                  (map split-to-ints)
                  (apply zip))
      race2 (->> input
                 (map #(str/replace % #"\D" ""))
                 (map parse-long))]

  (->> races1
       (map find-time)
       (apply *)
       (println "Part 1:"))
  
  (println "Part 2:" (find-time race2)))